jumanji/training/setup_train.py

```diff
@@ -393,20 +397,33 @@ def setup_training_state(
     params_state = agent.init_params(params_key)
 
     # Initialize environment states.
-    num_devices = jax.local_device_count()
+    num_local_devices = jax.local_device_count()
+    num_global_devices = jax.device_count()
+    num_workers = num_global_devices // num_local_devices
     reset_keys = jax.random.split(reset_key, agent.total_batch_size).reshape(
-        (num_devices, agent.batch_size_per_device, -1)
+        (
+            num_workers,
+            num_local_devices,
+            agent.total_batch_size // num_global_devices,
+            -1,
+        )
+    )
+    reset_keys_per_worker = reset_keys[jax.process_index()]
+    env_state, timestep = jax.pmap(env.reset, axis_name="devices")(
+        reset_keys_per_worker
     )
-    env_state, timestep = jax.pmap(env.reset, axis_name="devices")(reset_keys)
 
     # Initialize acting states.
-    acting_key_per_device = jax.random.split(acting_key, num_devices)
+    acting_key_per_device = jax.random.split(acting_key, num_global_devices).reshape(
+        num_workers, num_local_devices, -1
+    )
+    acting_key_per_device_worker = acting_key_per_device[jax.process_index()]
```

Comment:
A bit confused by the naming, specifically "_device_worker" - "worker" comes higher in the hierarchy, so maybe it should be `_worker_device`?

Reply:
Done